src/middlewares/isAdmin.js

```diff
@@ -0,0 +1,18 @@
+import {verifyToken } from './auth.middleware';
+
+    export const isAdmin = async(req, res, next) => {
+
+        verifyToken(req, res, function(){
+            console.log(req.user.id);
+            if(req.user?.role !== 'admin'){
+                return res.status(401).json({
+                    status: 401,
+                    message: 'Only admin is allowed to perform this task',
+                    });
+            }else{
+                next();
+            }
+        });
+    
+    };
+    
```

Comment:
Let's have global middleware which will support protecting route to be accessed by a specified role

`You can implement this in auth.middle...`
![image](https://user-images.githubusercontent.com/51375428/218050532-4b0b9fc7-5ca9-4f83-85bc-9de4f99a70f0.png)

`Use this middleware in route after checking if the user is authenticated
![image](https://user-images.githubusercontent.com/51375428/218051106-436a88b7-0fe6-4a02-8dee-f080ef000bd5.png)